src/events/mod.ts

```diff
@@ -0,0 +1,62 @@
+import { SlackManifest } from "../manifest/mod.ts";
+import { ManifestCustomEventSchema } from "../manifest/manifest_schema.ts";
+import {
+  CustomEventDefinition,
+  DefineEventSignature,
+  ICustomEvent,
+} from "./types.ts";
+
+export const DefineEvent: DefineEventSignature = <
+  Def extends CustomEventDefinition,
+>(
+  definition: Def,
+) => {
+  return new CustomEvent(definition);
+};
+
+export class CustomEvent<Def extends CustomEventDefinition>
+  implements ICustomEvent {
+  public id: string;
+  public title: string | undefined;
+  public description: string | undefined;
+
+  constructor(
+    public definition: Def,
+  ) {
+    this.id = definition.name;
+    this.definition = definition;
+    this.description = definition.description;
+    this.title = definition.title;
+  }
+
+  private generateReferenceString() {
+    return this.id;
+  }
+
+  toString() {
+    return this.generateReferenceString();
+  }
+
+  toJSON() {
+    return this.generateReferenceString();
+  }
+
```

Comment:
Is it intended for toString() and toJSON() to do the same thing?

Reply:
Yep! Overwriting these prototype methods is a pattern we follow elsewhere (like in types and workflows) to make sure that when a developer references the `CustomEvent` instance, we convert it to what our API endpoints expect for the manifest and API clients rather than the full object.

`toJSON()` supports the case where we convert an object into a string to pass to our APIs
`toString()`supports the case where devs may want to do something like `This is the event: ${MyEvent}`